src/renderer/src/routes/Onboarding/DeviceNamingScreen.tsx

```diff
@@ -0,0 +1,233 @@
+const DEVICE_NAME_MAX_BYTES = 512
+
+export const m = defineMessages({
+	title: {
+		id: 'screens.DeviceNamingScreen.title',
+		defaultMessage: 'Name Your Device',
+	},
+	description: {
+		id: 'screens.DeviceNamingScreen.description',
+		defaultMessage:
+			'A device name allows others using CoMapeo to invite you to projects.',
+	},
+	placeholder: {
+		id: 'screens.DeviceNamingScreen.placeholder',
+		defaultMessage: 'Device Name',
+	},
+	addName: {
+		id: 'screens.DeviceNamingScreen.addName',
+		defaultMessage: 'Add Name',
+	},
+	characterCount: {
+		id: 'screens.DeviceNamingScreen.characterCount',
+		defaultMessage: '{count}/{maxLength}',
+	},
+	errorSavingDeviceName: {
+		id: 'screens.DeviceNamingScreen.errorSavingDeviceName',
+		defaultMessage:
+			'An error occurred while saving your device name. Please try again.',
+	},
+	saving: {
+		id: 'screens.DeviceNamingScreen.saving',
+		defaultMessage: 'Saving...',
+	},
+})
+
+export const Route = createFileRoute('/Onboarding/DeviceNamingScreen')({
+	component: DeviceNamingScreenComponent,
+})
+
+const Container = styled('div')({
+	display: 'flex',
+	flexDirection: 'column',
+	alignItems: 'center',
+	justifyContent: 'center',
+	height: '100%',
+	backgroundColor: DARK_COMAPEO_BLUE,
+})
+const ContentBox = styled('div')({
+	backgroundColor: 'rgba(255, 255, 255, 0.94)',
+	border: `1px solid ${BLUE_GREY}`,
+	borderRadius: 8,
+	padding: 20,
+	width: '55%',
+	textAlign: 'center',
+	boxShadow: '0px 4px 4px 0px rgba(0, 0, 0, 0.02)',
+})
+const BodyTextWrapper = styled('div')({
+	maxWidth: '45%',
+	margin: '16px auto 0',
+	textAlign: 'center',
+})
+const ButtonContainer = styled('div')({
+	display: 'flex',
+	justifyContent: 'center',
+	gap: 15,
+	marginTop: 63,
+	padding: '0 20px',
+})
+const StyledImage = styled('img')({
+	marginBottom: 20,
+	width: 60,
+	height: 48,
+})
+const InputWrapper = styled('div')({
+	marginTop: 24,
+	marginBottom: 160,
+	display: 'flex',
+	flexDirection: 'column',
+	alignItems: 'center',
+})
+const StyledTextField = styled(TextField)({
+	width: '100%',
+	maxWidth: '400px',
+	backgroundColor: WHITE,
+	marginTop: 20,
+})
+const CharacterCount = styled(Text, {
+	shouldForwardProp: (prop) => prop !== 'error',
+})<{ error: boolean }>`
+	margin-top: 8px;
+	color: ${({ error }) => (error ? RED : BLACK)};
+	width: 100%;
+	max-width: 400px;
+	text-align: right;
+`
+
+export function DeviceNamingScreenComponent() {
+	const navigate = useNavigate()
+	const { formatMessage } = useIntl()
+	const [deviceName, setDeviceName] = useState('')
+	const [error, setError] = useState(false)
+	const [errorMessage, setErrorMessage] = useState('')
+	const setDeviceNameMutation = useEditDeviceInfo()
+
+	function getGraphemeSegments(text: string): Array<string> {
+		if (typeof Intl !== 'undefined' && 'Segmenter' in Intl) {
+			const segmenter = new Intl.Segmenter(undefined, {
+				granularity: 'grapheme',
+			})
+			const segments = [...segmenter.segment(text)].map((s) => s.segment)
+			return segments
```

Comment:
We only ever care about the number of graphemes. Rather than create an array just to get its length, could we change this function to `getGraphemeCount` and do something like this?

```ts
function countGraphemes(text: string): number {
  const segmenter = new Intl.Segmenter(undefined, {
    granularity: 'grapheme',
  })
  let result = 0
  for (const _ of segmenter.segment(text)) result++
  return result
}
```